src/components/encounter/encounter-form.component.tsx

```diff
@@ -172,6 +173,13 @@ const EncounterForm: React.FC<EncounterFormProps> = ({
     formFieldHandlers,
   );
 
+  useEffect(() => {
+    getEncounter(formJson).then((data) => {
+      setEncounter(data.encounter);
+      setIsLoadingEncounter(data.isLoading);
+    });
+  }, [formJson]);
```

Comment:
Would it be a bad thing to just refactor the hook and we keep:

```ts
  const { encounter, isLoading: isLoadingEncounter } = useEncounter(formJson);

```

Reply:
Are u suggesting using a similar pattern as the resolvers. With the useEffect in the function?

Reply:
Kind of

Reply:
Done